lib/cn/vblock_builder.c

```diff
@@ -234,20 +337,22 @@ vbb_add_entry(
     assert(vlen);
     assert(vlen <= HSE_KVS_VALUE_LEN_MAX);
 
-    if (HSE_UNLIKELY(!_vblock_has_room(bld, vlen))) {
-        err = _vblock_finish(bld);
+    bld->cur_kobj = kobj;
```

Comment:
This doesn't feel safe.  It stores pointers to callers memory, which could point into compaction read buffers that are reused by the time the key is copied out in vbb_finish().  Why not copy the key here and store in the builder?